src/doctopus/storage/impls/temp_fs.clj

```diff
@@ -0,0 +1,33 @@
+(ns doctopus.storage.impls.temp-fs
+  (:require [clojure.string :as str]
+            [doctopus.files :as files]
+            [doctopus.storage.impls.fs-impl :refer [save-html-file] :as fs-impl]
+            [me.raynes.fs :as fs]))
+
+;; The root of the temp filesystem. Each Thing will store its stuff
+;; within this directory
+(def temp-dir (atom (fs/temp-dir  "doctopus-temp")))
+
+(defn regenerate-temp-dir
+  "Generates a new temp dir"
+  []
+  (swap! temp-dir fs/temp-dir "doctopus-temp"))
+
+(defn save-fn
+  [key path-html-pairs]
+  (doseq [[path html] path-html-pairs] (save-html-file @temp-dir key path html)))
+
+(defn load-fn
+  "This is as Version 1 a way of doing this as I can think of:
+
+  1. We load html
+  2. We wrap that html in a function that returns itself, so we can Bidi"
+  [key]
+  (binding [fs/*cwd* @temp-dir]
```

Comment:
`binding` re-defines a `var` within a context. So, for instance: the `fs` package has a state `var` called `*cwd*` which, globally, contains the directory the JVM was started in. It automatically splices `*cwd*` in to things like relative paths of files (`(fs/file "foo.txt"`) will automatically get `*cwd*` appended to the front.) Binding lets us change, locally and temporarily, the value of that var, allowing a JVM to immitate a notion of current working directory :P